portable/GCC/ARM_CA9/portASM.S

```diff
@@ -201,12 +201,13 @@ FreeRTOS_IRQ_Handler:
 	PUSH	{r0-r4, lr}
 	LDR		r1, vApplicationIRQHandlerConst
 	BLX		r1
-	POP		{r0-r4, lr}
-	ADD		sp, sp, r2
-
+	
 	CPSID	i
 	DSB
 	ISB
+	
+	POP		{r0-r4, lr}

```

Comment:
Grateful if you can explain why this change is needed.

Its been a while since I worked on this port, but as I recall these registers are pushed and popped to retain their values across a function call, rather than to protect them from interrupts, hence they are popped from outside of the critical section (i.e. before interrupts are disabled).  In the current release code version, if an interrupt occurred between the registers being popped and interrupts being disabled then the CPU would switch to IRQ mode, causing it to switch from the SYSTEM mode stack to the IRQ mode stack, and so not clobber these register values.

Maybe you are highlighting a problem in the interrupt nesting code, if so, please provide an explanation of the sequence that could cause a problem.  The registers are not protected from interrupts occurring before the function call, or within the function call, so I'm not sure why they need protecting after the function call.